katpoint/projection.py

```diff
@@ -303,13 +480,12 @@ def sphere_to_plane_tan(az0, el0, az, el):
 
     Raises
     ------
-    ValueError
-        If input values are out of range, or target is too far from reference
-
+    OutOfRangeError
+        If an elevation is out of range or target is too far from reference,
+        and out-of-range treatment is 'raise'
     """
-    ortho_x, ortho_y, cos_theta = sphere_to_ortho(az0, el0, az, el)
-    if np.any(cos_theta <= 0.0):
-        raise ValueError('Target point pi/2 radians or more away from reference point')
+    # Angular separation theta must be strictly < 90 degrees - pick 1e-6 radians less
+    ortho_x, ortho_y, cos_theta = sphere_to_ortho(az0, el0, az, el, min_cos_theta=1e-6)
```

Comment:
From the comment of fd70079f6:
```
    Replace all coordinate checks and subsequent ValueError raises with
    `OutOfRange.treat()`. All conditions are faithfully preserved, except
    one in sphere_to_plane_tan. There cos(theta) can't be 0, so that the
    valid range is an open interval. Since OutOfRange uses np.clip which
    enforces a closed interval, move the threshold slightly above 0.
    I picked 1e-5 due to its popularity in AIPS DIRCOS and NEWPOS, the
    original source of these routines.```

From the comment of 73bc7527:
```
    Improve the threshold in one instance (sphere_to_plane_tan)
    so that one can now get within 0.2" of the domain boundary. This is an
    arbitrary choice that results in (x, y) values of up to ~1e6 (they
    blow up at the boundary).
```

I introduced an arbitrary strictly positive threshold because the out-of-range machinery expects a closed valid interval and for TAN this is an open interval.

I could bring this closer to eps if it helps.

Reply:
Leaving this for now (it's a separate issue from SR-1511).